devices/HumanDataCollector/HumanDataCollector.cpp

```diff
@@ -0,0 +1,663 @@
+
+    // Joint Quantities
+    size_t stateNumberOfJoints;
+    std::vector<std::string> stateJointNames;
+    std::vector<double> jointPositions;
+    std::vector<double> jointVelocities;
+
+    // CoM Quantities
+    std::array<double, 3> comPosition;
+    std::array<double, 3> comVelocity;
+    std::array<double, 6> comProperAccInBaseFrame;
+    std::array<double, 6> comProperAccInWorldFrame;
+
+    // Wrench Measurements
+    size_t numberOfWrenchMeasurementSources;
+    std::vector<std::string> wrenchMeasurementSourceNames;
+    std::vector<double> wrenchMeasurementValues;
+
+    // Wrench Estimates
+    size_t numberOfWrenchEstimateSources;
+    std::vector<std::string> wrenchEstimateSourceNames;
+    std::vector<double> wrenchEstimateValues;
+
+    // Joint Torques
+    size_t dynamicsNumberOfJoints;
+    std::vector<std::string> dynamicsJointNames;
+    std::vector<double> jointTorques;
+
+    // Yarp ports for streaming data from IHumanState interface of HumanStateProvider
+    yarp::os::BufferedPort<yarp::sig::Vector> basePoseDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> baseVelocityDataPort;
+    yarp::os::BufferedPort<yarp::os::Bottle> stateJointNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> jointPositionsDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> jointVelocitiesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comPositionDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comVelocityDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comProperAccelerationInBaseFrameDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> comProperAccelerationInWorldFrameDataPort;
+
+    // Yarp ports for streaming data from IHumanWrench interface of HumanWrenchProvider
+    yarp::os::BufferedPort<yarp::os::Bottle> wrenchMeasurementSourceNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> wrenchMeasurementValuesDataPort;
+
+    // Yarp ports for streaming data from IHumanWrench interfce and IHumanDynamics interface of HumanDynamicsEstimator
+    yarp::os::BufferedPort<yarp::os::Bottle> wrenchEstimateSourceNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> wrenchEstimateValuesDataPort;
+
+    yarp::os::BufferedPort<yarp::os::Bottle> dynamicsJointNamesDataPort;
+    yarp::os::BufferedPort<yarp::sig::Vector> jointTorquesDataPort;
+};
+
+HumanDataCollector::HumanDataCollector()
+    : PeriodicThread(DefaultPeriod)
+    , pImpl{new impl()}
+{}
+
+HumanDataCollector::~HumanDataCollector() {}
+
+bool HumanDataCollector::open(yarp::os::Searchable &config) {
+
+    // ===============================
+    // CHECK THE CONFIGURATION OPTIONS
+    // ===============================
+
+    if (!(config.check("period") && config.find("period").isFloat64())) {
+        yInfo() << LogPrefix << "Using default period";
+    }
+
+    if (!(config.check("portPrefix") && config.find("portPrefix").isString())) {
+        yInfo() << LogPrefix << "Using default portPrefix HDE ";
```

Comment:
I think that, if not specified, we can leave the ports without prefix like in 
https://github.com/robotology/human-dynamics-estimation/blob/fd6392ad7deb0de5579ba46abd6df60fd543b057/publishers/HumanStatePublisher/HumanStatePublisher.cpp#L186-L196

https://github.com/robotology/human-dynamics-estimation/blob/fd6392ad7deb0de5579ba46abd6df60fd543b057/publishers/HumanStatePublisher/HumanStatePublisher.cpp#L208-L217

Reply:
@lrapetti as discussed I will continue with having a prefix for the time being as we need to have `yarpscope` config files corresponding to the ports opened by `HumanDataCollector` device. In anycase, we need to do a clean up of all the devices in the library to have a consistent naming for the ports opened by different devices.